File: internal/parser/java/testdata/http_client.java
package com.example.client;

import org.springframework.web.client.RestTemplate;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.URI;

public class ApiClient {
    private RestTemplate restTemplate = new RestTemplate();

    public String getUser(String id) {
        return restTemplate.getForObject("/api/users/" + id, String.class);
    }

    public void createUser(String data) {
        restTemplate.postForEntity("/api/users", data, String.class);
    }

    public void deleteUser(String id) {
        restTemplate.delete("/api/users/" + id);
    }
}

File: internal/parser/golang/testdata/http_client.go
package client

import (
	"context"
	"net/http"
	"net/url"
	"strings"
)

func fetchUser(id string) {
	http.Get("/api/users/" + id)
}

func createUser(data string) {
	http.Post("/api/users", "application/json", strings.NewReader(data))
}

func headCheck() {
	http.Head("/health")
}

func postForm() {
	http.PostForm("/api/login", url.Values{"user": {"admin"}})
}

func customRequest(ctx context.Context) {
	req, _ := http.NewRequestWithContext(ctx, "PUT", "/api/users/123", nil)
	client := &http.Client{}
	client.Do(req)
}

func clientGet() {
	c := &http.Client{}
	c.Get("/api/items")
}

func clientPost() {
	c := http.Client{}
	c.Post("/api/items", "application/json", nil)
}
